DQMOffline/MuonDPG/plugins/plugins.cc

```diff
@@ -0,0 +1,21 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
```

Comment:
`MakerMacros` is included here, not needed to have it included also in every classes. However, I would rather:
 - Define DEFINE_FWK_MODULE inside every corresponding implementation file
 - Move all your plugins from /interface and /src to the /plugin area (unless there is a reason for not doing so)
 - Possibly also unify .h and .cc files there

Reply:
If I understand correctly this means that plugin.cc could be removed since plugins would be defined inside the corresponding .cc file, right?

Could you please also clarify what do you mean by "unify .h and .cc files"? Does this means moving both .h and .cc to DQMOffline/MuonDPG/plugins?

Reply:
Ciao Leonardo.

> @perrotta If I understand correctly this means that plugin.cc could be removed since plugins would be defined inside the corresponding .cc file, right?

Correct

> Could you please also clarify what do you mean by "unify .h and .cc files"? Does this means moving both .h and .cc to DQMOffline/MuonDPG/plugins?

This is not mandatory, but a good practice for plugins whose interface doesn't need to be exposed anywhere else is that you combine the .h and the .cc files into one singe .cc file, and put it in the \plugin area. This would allow you (also) to remove duplicate header includes between .h and .cc 

Alternatively, but not preferred, you can move both .h and .cc files in \plugin and let them unchanged.

In your case, BaseTnPEfficiencyTask could remain split into .h and .cc, because you need to link the interface into the derived classes: all those derived classes, which constitute the actual plugins, should preferably get coded into a single .cc file instead

Reply:
I'm going to follow your instructions: I'm removing `plugins.cc` and unify .h and .cc of plugin modules in /plugins.